web/profiles/custom/yalesites_profile/modules/custom/ys_views_basic/src/Plugin/views/filter/ViewsBasicFilter.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Drupal\ys_views_basic\Plugin\views\filter;
+
+use Drupal\views\Plugin\views\filter\FilterPluginBase;
+
+/**
+ * Filter that receives JSON overrides from a Views Basic field.
+ *
+ * @ingroup views_filter_handlers
+ *
+ * @ViewsFilter("views_basic_filter")
+ */
+class ViewsBasicFilter extends FilterPluginBase {
+
+  /**
+   * Add this filter to the query.
+   */
+  public function query() {
+    // Ensure the main table for this handler is in the query.
+    $this->ensureMyTable();
+
+    // Parse content type filters.
+    // @todo Add taxonomy term as a filter. Code here: https://gist.github.com/marcb4k/fd4716eb133aa846f90a9e42c80def8c.
+    foreach ($this->value['filters']['types'] as $content_type) {
+      $this->query->addWhere($this->options['group'], 'type', $content_type);
+    }
```

Comment:
```suggestion
    // Ensure the main table for this handler is in the query.
    $this->ensureMyTable();

    /** @var \Drupal\views\Plugin\views\query\Sql $query */
    $query = $this->query;

    // Parse content type filters.
    // @todo Add taxonomy term as a filter. Code here: https://gist.github.com/marcb4k/fd4716eb133aa846f90a9e42c80def8c.
    foreach ($this->value['filters']['types'] as $content_type) {
      $query->addWhere($this->options['group'], 'type', $content_type);
    }
```

Reply:
Just and FYI. If you want to appease the IDE overlords, then this is the common way to do it. The reason `addWhere` is not recognized is that not all view can follow this syntax. I think SOLR data has a different set of methods for querying data. I'm pretty sure external entities from a webservice don't explicitly use a where clause. 

We know that all YaleSites database data will be a flavor of SQL. The IDE is just barking because it technically does not have to be true in Drupal.